transaction-context/src/lib.rs

```diff
@@ -884,7 +900,8 @@ pub struct BorrowedAccount<'a> {
     transaction_context: &'a TransactionContext,
     instruction_context: &'a InstructionContext,
     index_in_transaction: IndexOfAccount,
-    index_in_instruction: IndexOfAccount,
+    // Program accounts are not part of the instruction_accounts vector
```

Comment:
and thus `None`

Reply:
Fixed in https://github.com/anza-xyz/agave/pull/7220/commits/cbc0c0b09b77dd53d386887a2c47fde920270a83